ui/coordinateTrainer/src/ctrl.ts

```diff
@@ -126,16 +175,20 @@ export default class CoordinateTrainerCtrl {
     this.playing = true;
     this.hasPlayed = true;
     this.score = 0;
-    this.timeLeft = DURATION;
     this.currentKey = '';
     this.nextKey = '';
 
+    console.debug(this.timeDisabled);
+
     // Redraw the chessground to remove the resize handle
     this.chessground?.redrawAll();
 
     // In case random is selected, recompute orientation
     this.setOrientation(orientationFromColorChoice(this.colorChoice));
 
+    // Custom duration
+    this.setDuration(durationFromTimeControlChoice(this.timeControlChoice));
```

Comment:
If I understand correctly, this line could just become:
```typescript
this.timeLeft = this.duration;
```

And since the line above (`this.setOrientation`) includes a `this.redraw()`, you probably want to move this line before that.

Reply:
Thank you very much!